### DISCUSSION TITLE
Integrating RNE theming with React Navigation

### DISCUSSION THREAD

**Discussion by Ravi-Bossema** [react-native-elements/react-native-elements#3598] [Q&A] [answered]
I recently started using version: 4.0.0-rc.5 of RNE and I'm really loving the idea of the theming provided by it. 
However, it doesn't seem to work very well with the theming in React Navigation (or at least the `DrawerNavigator` of React navigation). 
When trying something like this:
```jsx
import React from 'react';
import { ThemeProvider, createTheme  } from '@rneui/themed';
import { NavigationContainer } from '@react-navigation/native';
import MyDrawerNavigator from './routers/MyDrawerNavigator';

const myTheme = createTheme({
	lightColors: {
            ...
	},
	darkColors: {
            ...
	},
});

export default function App() {
	return (
		<ThemeProvider theme={myTheme}>
			<NavigationContainer theme={myTheme}>
				<MyDrawerNavigator />
			</NavigationContainer>
		</ThemeProvider>
	);
}
```
I get a render error saying `undefined is not an object (evaluating 'colors.card')`.
Makes sense since the RNE theme object does not contain a color propery called `card`.
I hoped that adding a `card` property to `myTheme` would solve it:

```jsx
const myTheme = createTheme({
	lightColors: {
            card:  ' ... '
            ...
	},
	darkColors: {
            card:  ' ... '
            ...
	},
});
```
But this doesn't solve the problem. Likely because the `DrawerNavigator` can't access the `lightColors` and `darkColors` properties, which raises a question: Is there an elegant way of using the RNE theming with React Navigation? 

I would like to have 1 central theme with light and dark colors that are used by RNE components as well as React Navigation components.

**Comment by Ravi-Bossema** [comment, marked as answer]
I think I found a good solution!

By taking the `NavigationContainer` out of App.js you're able to use RNE's `useTheme` to get the current theme and pass it on to the `NavigationContainer` since the `useTheme` hook  returns a theme object with a `colors` property. This `colors` property is derived from the `lightColors` and `darkColors` properties based on whether you're in light mode or in dark mode. This way the solution I had in mind earlier does work! In the end it looks something like this:

App.js
```jsx
import React from 'react';
import { ThemeProvider, createTheme  } from '@rneui/themed';
import MyDrawerNavigator from './routers/MyDrawerNavigator';

const myTheme = createTheme({
	lightColors: {
            card:  ' ... '
            ...
	},
	darkColors: {
            card:  ' ... '
            ...
	},
});

export default function App() {
	return (
		<ThemeProvider theme={myTheme}>
			<MyDrawerNavigator />
		</ThemeProvider>
	);
}
```

MyDrawerNavigator.jsx
```jsx
import React from 'react';
import { createDrawerNavigator } from '@react-navigation/drawer';
import { useTheme } from '@rneui/themed';
import { NavigationContainer } from '@react-navigation/native';

const Drawer = createDrawerNavigator();

function MainRouter() {
	const { theme } = useTheme();
	return (
		<NavigationContainer theme={theme}>
			<Drawer.Navigator>
				...
			</Drawer.Navigator>
		</NavigationContainer>
	);
}

export default MainRouter;

```

Do note that not only the `card` property is missing, the React Navigation theme object also uses `text`, `border`, and `notification` properties that are not in the default RNE theme object, so you'd have to define those too where you create your theme.

**Reply by arpitBhalla** [reply]
Hey, you can extend the theme, please refer documentation
